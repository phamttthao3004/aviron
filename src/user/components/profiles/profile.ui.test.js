import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ProfileUI from './profile.ui';
import CONSTANTS from '../../scripts/apps/constants';

describe('Profile', () => {
    it(`should render a list with 4 items by default`, () => {
        const props = { currentTab: undefined, clickElement: jest.fn() };
        const wrapper = shallow(<ProfileUI {...props} />);
        const li = wrapper.find('li');
        expect(li.length).toBe(4);
    });
    it('Render Profile UI component with currentTab=undefined', () => {
        const props = { currentTab: undefined, clickElement: undefined };
        const wrapper = shallow(<ProfileUI {...props} />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
    it('Render Profile UI component with currentTab=PROFILE_OVERVIEW', () => {
        const props = { currentTab: CONSTANTS.PROFILE_PAGE.CURRENT_TAB.PROFILE_OVERVIEW, clickElement: undefined };
        const wrapper = shallow(<ProfileUI {...props} />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
    it('Render Profile UI component with currentTab=PROFILE_WORKOUT', () => {
        const props = { currentTab: CONSTANTS.PROFILE_PAGE.CURRENT_TAB.PROFILE_WORKOUT, clickElement: undefined };
        const wrapper = shallow(<ProfileUI {...props} />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
    it('Render Profile UI component with currentTab=PROFILE_ACHIEVEMENT', () => {
        const props = { currentTab: CONSTANTS.PROFILE_PAGE.CURRENT_TAB.PROFILE_ACHIEVEMENT, clickElement: undefined };
        const wrapper = shallow(<ProfileUI {...props} />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
    it('Render Profile UI component with currentTab=PROFILE_BASICINFO', () => {
        const props = { currentTab: CONSTANTS.PROFILE_PAGE.CURRENT_TAB.PROFILE_BASICINFO, clickElement: undefined };
        const wrapper = shallow(<ProfileUI {...props} />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});
