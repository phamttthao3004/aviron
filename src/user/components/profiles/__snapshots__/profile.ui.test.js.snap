// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Render Profile UI component with currentTab=PROFILE_ACHIEVEMENT 1`] = `
<div>
  <div
    className="container-fluid"
  >
    <ul
      data-testid="tab-list"
    >
      <li
        className="pointer "
        onClick={[Function]}
        role="presentation"
      >
        <span>
          OVERVIEW
        </span>
      </li>
      <li
        className="pointer "
        onClick={[Function]}
        role="presentation"
      >
        <span>
          WORKOUTS
        </span>
      </li>
      <li
        className="pointer active"
        onClick={[Function]}
        role="presentation"
      >
        <span>
          ACHIEVEMENTS
        </span>
      </li>
      <li
        className="pointer "
        onClick={[Function]}
        role="presentation"
      >
        <span>
          BASIC INFO
        </span>
      </li>
    </ul>
  </div>
  <ProfileUserInfo />
</div>
`;

exports[`Profile Render Profile UI component with currentTab=PROFILE_BASICINFO 1`] = `
<div>
  <div
    className="container-fluid"
  >
    <ul
      data-testid="tab-list"
    >
      <li
        className="pointer "
        onClick={[Function]}
        role="presentation"
      >
        <span>
          OVERVIEW
        </span>
      </li>
      <li
        className="pointer "
        onClick={[Function]}
        role="presentation"
      >
        <span>
          WORKOUTS
        </span>
      </li>
      <li
        className="pointer "
        onClick={[Function]}
        role="presentation"
      >
        <span>
          ACHIEVEMENTS
        </span>
      </li>
      <li
        className="pointer active"
        onClick={[Function]}
        role="presentation"
      >
        <span>
          BASIC INFO
        </span>
      </li>
    </ul>
  </div>
  <ProfileUserInfo />
</div>
`;

exports[`Profile Render Profile UI component with currentTab=PROFILE_OVERVIEW 1`] = `
<div>
  <div
    className="container-fluid"
  >
    <ul
      data-testid="tab-list"
    >
      <li
        className="pointer active"
        onClick={[Function]}
        role="presentation"
      >
        <span>
          OVERVIEW
        </span>
      </li>
      <li
        className="pointer "
        onClick={[Function]}
        role="presentation"
      >
        <span>
          WORKOUTS
        </span>
      </li>
      <li
        className="pointer "
        onClick={[Function]}
        role="presentation"
      >
        <span>
          ACHIEVEMENTS
        </span>
      </li>
      <li
        className="pointer "
        onClick={[Function]}
        role="presentation"
      >
        <span>
          BASIC INFO
        </span>
      </li>
    </ul>
  </div>
  <ProfileUserInfo />
</div>
`;

exports[`Profile Render Profile UI component with currentTab=PROFILE_WORKOUT 1`] = `
<div>
  <div
    className="container-fluid"
  >
    <ul
      data-testid="tab-list"
    >
      <li
        className="pointer "
        onClick={[Function]}
        role="presentation"
      >
        <span>
          OVERVIEW
        </span>
      </li>
      <li
        className="pointer active"
        onClick={[Function]}
        role="presentation"
      >
        <span>
          WORKOUTS
        </span>
      </li>
      <li
        className="pointer "
        onClick={[Function]}
        role="presentation"
      >
        <span>
          ACHIEVEMENTS
        </span>
      </li>
      <li
        className="pointer "
        onClick={[Function]}
        role="presentation"
      >
        <span>
          BASIC INFO
        </span>
      </li>
    </ul>
  </div>
  <ProfileUserInfo />
</div>
`;

exports[`Profile Render Profile UI component with currentTab=undefined 1`] = `
<div>
  <div
    className="container-fluid"
  >
    <ul
      data-testid="tab-list"
    >
      <li
        className="pointer "
        onClick={[Function]}
        role="presentation"
      >
        <span>
          OVERVIEW
        </span>
      </li>
      <li
        className="pointer "
        onClick={[Function]}
        role="presentation"
      >
        <span>
          WORKOUTS
        </span>
      </li>
      <li
        className="pointer "
        onClick={[Function]}
        role="presentation"
      >
        <span>
          ACHIEVEMENTS
        </span>
      </li>
      <li
        className="pointer "
        onClick={[Function]}
        role="presentation"
      >
        <span>
          BASIC INFO
        </span>
      </li>
    </ul>
  </div>
  <ProfileUserInfo />
</div>
`;
